PiwikTracker/Dispatcher.swift

```diff
@@ -4,7 +4,9 @@ public protocol Dispatcher {
     
     var baseURL: URL { get }
     
-    var userAgent: String? { get }
+    var userAgent: String? { get set }
+    
+    func setUserAgent(ua: String)
```

Comment:
I guess most users will never need to change the user agent.

Normally I would try to avoid exposing functionality in a public api (especially protocols) that is almost never used. Once you exposed it you need to test it and maintain it forever. 

Then it is better to allow those special users to extend the code or inject their own implementations to meet their goals. One such way is to allow the dispatcher to be instantiated with a customised user agent or maybe create a specialised dispatcher implementation (that inherits or wrap) the default implementation and offers a set or append method. This is still possible without exposing this functionality in the protocol.